### DISCUSSION TITLE
Add support for specifying folders that aren't namespace providers.

### DISCUSSION THREAD

**Discussion by tacosontitan** [dotnet/roslyn#71956] [Ideas]
[IDE0130](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0130) helps keep namespaces equivalent to the folder structure, and I completely agree with it. However, on occasion, I have a folder that exists purely for organizational benefit and should not be part of the namespace. A good example is the framework project I maintain at work where we define *many* extensions for `IServiceCollection` as part of our `Runtime.DependencyInjection` namespace to support our ecosystem, and these are broken down into separate files:

![image](https://github.com/dotnet/roslyn/assets/65432314/8a51539c-9002-42bf-a1f1-d58da6b0d184)

Each of these files is defined as:

```cs
namespace Runtime.DependencyInjection;

public static partial class ServiceCollectionExtensions
{
    ...
}
```

Unfortunately for this use case, according to IDE0130, the namespace should be adjusted to add the `SeviceCollectionExtensions` folder effectively becoming `Runtime.DependencyInjection.ServiceCollectionExtensions`; this forces us to import two namespaces when working with our dependency injection extensions simply because of an organizational decision we made.

JetBrains products have a way for us to specify that a folder isn't a namespace provider, and I'd love to be capable of specifying that a folder is not a namespace provider at the `.csproj` level, so something like:

```xml
<ItemGroup Label="Pardoned Namespace Violators">
    <NamespaceProvider "DependencyInjection/ServiceCollectionExtensions">false</NamespaceProvider>
</ItemGroup>
```

I initially thought maybe `Folder` or `None` could be used with an additional property like:

```xml
<ItemGroup Label="Pardoned Namespace Violators">
    <Folder Include="DependencyInjection/ServiceCollectionExtensions" NamespaceProvider="False"/>
</ItemGroup>
```

However, given that I don't have enough experience with those yet, please take that with a grain of salt; this discussion is mostly to bring the idea forward for community feedback.

**Comment by davidwengier** [comment]
Related (or dupe of?): https://github.com/dotnet/project-system/issues/5372
Also: https://github.com/dotnet/roslyn/issues/45494
